pom.xml

```diff
@@ -109,6 +109,18 @@
             <version>4.12</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.16.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <version>4.0.2</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
I left these in from my original tests, the tests can be rewritten to use standard junit assert functionality instead of third party libraries, I didn't want to block creating the initial PR on it.